### PR TITLE
improved version of "fixVersions.sh"

### PR DESCRIPTION
**Pull request by pngowda**
@kthoms @cdietrich  
Hi 
Here is the improved version of "fixVersionsScript.sh" .
could you please review this pull request. and let me know if some changes required.
Regards,
Prajwal

### CODE REVIEW

**Comment by cdietrich** [comment]
@pngowda can you please signoff the commit?

**Comment by kthoms** [comment]
For me the script takes long (>1min) when executed:
```
./fixVersions.sh -f 2.15.0
# from_version: 2.15.0
# to_version:
# 'to_version' is unset or set to the empty string
# deriving 'to_version' from 'from_verison'
# to_version: 2.16.0
# OS: Mac OS
```

It creates files with double quote at the end:

screenshot 94

File: fixVersions.sh [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  darwin*)  echo "# OS: Mac OS" 
+            xargs_sed_inplace="xargs sed -i ''";; 
+  linux*)   echo "# OS: LINUX"
+            xargs_sed_inplace="xargs sed -i";;
+  *)        echo "# OS: $OSTYPE"
+            xargs_sed_inplace="xargs sed -i";;
+esac
+
+
+find ../ -type f -name "MANIFEST.MF" | $xargs_sed_inplace -e "s/${from}.qualifier/${to}.qualifier/g"
```
Comment (tivervac) [inline fixVersions.sh]: Doesn't the relative path make it dependent on where you execute this script from?
e.g. different results when executing `./xtext/git/xtext-umbrella/fixVersions.sh` from `~` than executing `./fixVersions.sh` from `~/xtext/git/xtext-umbrella/`?

Some solutions to this mentioned here https://stackoverflow.com/questions/630372/determine-the-path-of-the-executing-bash-script

**Comment by kthoms** [comment]
Why was the PR closed?

**Comment by pngowda** [comment]
@kthoms @tivervac 
could you please review.
have done some fixes as per the review comment. I use linux as my native OS, in that it the run time is not more than 2 sec. but in Mac its taking more time I have no clue.

**Comment by cdietrich** [comment]
hmm iplog bot is in a bad state again :(

**Comment by cdietrich** [comment]
https://bugs.eclipse.org/bugs/show_bug.cgi?id=540008

**Comment by tivervac** [comment]
@pngowda LGTM

**Comment by kthoms** [comment]
Also I'd expect that a development version would be specified with `-SNAPSHOT` suffix. How to distinguish between the dev and release version otherwise? For certain pom.xml the `-SNAPSHOT` must be removed for the final release, for others (used for Tycho builds) it has to stay.

**Comment by kthoms** [comment]
That said, I expect that it should be valid to execute with `fixVersions.sh -f 2.16.0-SNAPSHOT -t 2.16.0.M1` or `fixVersions.sh -f 2.16.0.RC1 -t 2.16.0`

**Comment by cdietrich** [comment]
I would expect two scripts for that. I don’t think that can be done with one

**Comment by cdietrich** [comment]
I created the initial version exactly for the snapshot secede. Nothing else. Prajwal added the ask for versions you wanted to have

**Comment by kthoms** [comment]
The script only works in the umbrella repo. When I execute it in the directory of xtext-core, nothing happens. I miss a bit feedback which files are actually processed.

**Comment by cdietrich** [comment]
@kthoms  see https://github.com/eclipse/xtext-umbrella/pull/84#pullrequestreview-163469741

**Comment by pngowda** [comment]
@kthoms @cdietrich 

I discussed with Christian some time ago and I got to know that, there are different cases here.
>snapshot to snapshot
>milestone to snapshot
>milestone to relase
>release to snapshot
etc...

right now the script handles snapshot to snapshot(initial version from Christian) so, we need to finalize on these cases and define the set of files to be changed accordingly. then it would be easy for me to implement the script as generic as possible.

**Comment by szarnekow** [comment]
Is there still progress being made on this PR?

**Comment by pngowda** [comment]
right now it handles Snapshot to Snapshot bumping.need to define set of files to be changed for other cases.

**Comment by cdietrich** [comment]
@kthoms please review

**Comment by kthoms** [comment]
Did not work (macOS)
```
./fixVersions.sh -f 2.16.0
# from_version: 2.16.0
# to_version:
# bump_to:
# 'to_version' is unset or set to the empty string
# deriving 'to_version' from 'from_verison'
# to_version: 2.17.0
./fixVersions.sh: line 121: [: ==: unary operator expected
```

File: fixVersions.sh [hunk 2]
```diff
@@ -3,10 +3,14 @@
 usage ()
 {
   echo "----------------------------------------------------------"
-  echo "Usage   : fixVersions.sh -f <from_version> [-t <to_version>]"
-  echo "Example : fixVersions.sh -f '2.15.0' -t '2.16.0'"
-  echo "Example : fixVersions.sh -f '2.15.0' "
+  echo "Usage   : fixVersions.sh -f <from_version> [-t <to_version>] [-b <StoS|StoR|MtoR|RCtoR>]"
```
[the remark this answers is not in the document]
Comment (pngowda) [reply fixVersions.sh]: its not optional. i have fixed it

File: fixVersions.sh [hunk 3]
```diff
@@ -70,26 +118,20 @@ fi
...
 checkVersionFormat "$to"
 
-xargs_sed_inplace() {
-	if [[ "$OSTYPE" == "darwin"* ]]; then
-		xargs sed -i '' "$@"
-	else
-		xargs sed -i "$@" 
-	fi	
-}
+if [ $bump == 'StoS' ]; then
```
[the remark this answers is not in the document]
Comment (pngowda) [reply fixVersions.sh]: its been taken care now. well, it works on linux OS anyway

**Comment by cdietrich** [comment]
@kthoms please reexamine

**Comment by kthoms** [comment]
Problem was usage of the `currentDirectory` property. This can be avoided when switching to the repository directory and then execute find on the current directory `.`. Makes the code also easier.

**Pull request merged**